source/server/http/admin.h

```diff
@@ -252,6 +283,8 @@ class AdminFilter : public Http::StreamDecoderFilter, Logger::Loggable<Logger::I
   AdminImpl& parent_;
   Http::StreamDecoderFilterCallbacks* callbacks_{};
   Http::HeaderMap* request_headers_{};
+  std::unique_ptr<std::list<std::function<void()>>> on_destroy_callbacks_ =
```

Comment:
nit: I don't think it's really worth optimizing this to create the list when needed, as this is admin code. I would just remove the unique_ptr here and always have the list.

Reply:
Sure? Not sure it will have any use beside in the Hystrix handler.

Reply:
I don't feel very strongly about it, but this makes the code harder to read and I don't think the slightly smaller CPU/memory use matters here.